raijin_scripts/execute_s2_to_s3_rolling/s2_to_s3_rolling.py

```diff
@@ -0,0 +1,166 @@
+            return False
+    else:
+        return True
+
+
+def sync_granule(granule, s3_bucket):
+    local_path = os.path.join(NCI_DIR, granule)
+    s3_path = "s3://{s3_bucket}/{s3_path}/{granule}".format(
+        s3_bucket=s3_bucket,
+        s3_path=S3_PATH,
+        granule=granule
+    )
+
+    # Remove any data that shouldn't be there and exclude the metadatta and NBART
+    command = "aws s3 sync {local_path} {s3_path} --delete --exclude NBART/* --exclude ARD-METADATA.yaml".format(
+        local_path=local_path,
+        s3_path=s3_path
+    )
+
+    return_code = subprocess.call(command, shell=True)
+
+    # If the return code is zero, we have success.
+    return return_code == 0
+
+
+def replace_metadata(granule, s3_bucket, s3_metadata_path):
+    s3 = boto3.resource("s3").Bucket(s3_bucket)
+
+    yaml_file = "{nci_path}/{granule}/ARD-METADATA.yaml".format(
+        nci_path=NCI_DIR,
+        granule=granule
+    )
+
+    with open(yaml_file) as config_file:
+        temp_metadata = yaml.load(config_file, Loader=yaml.CSafeLoader)
+
+    del temp_metadata['image']['bands']['nbart_blue']
+    del temp_metadata['image']['bands']['nbart_coastal_aerosol']
+    del temp_metadata['image']['bands']['nbart_contiguity']
+    del temp_metadata['image']['bands']['nbart_green']
+    del temp_metadata['image']['bands']['nbart_nir_1']
+    del temp_metadata['image']['bands']['nbart_nir_2']
+    del temp_metadata['image']['bands']['nbart_red']
+    del temp_metadata['image']['bands']['nbart_red_edge_1']
+    del temp_metadata['image']['bands']['nbart_red_edge_2']
+    del temp_metadata['image']['bands']['nbart_red_edge_3']
+    del temp_metadata['image']['bands']['nbart_swir_2']
+    del temp_metadata['image']['bands']['nbart_swir_3']
+    del temp_metadata['lineage']
+    temp_metadata['creation_dt'] = temp_metadata['extent']['center_dt']
+    temp_metadata['product_type'] = 'S2MSIARD_NBAR'
+
+    # Create dataset ID based on Kirill's magic
+    temp_metadata['id'] = str(odc_uuid("s2_to_s3_rolling", "1.0.0", [temp_metadata['id']]))
+
+    # Write to S3 directly
+    s3.Object(key=s3_metadata_path).put(Body=yaml.dump(
+        temp_metadata, default_flow_style=False, Dumper=yaml.CSafeDumper)
+    )
+
+
+def sync_dates(num_days, s3_bucket, end_date, update=False):
+    # Since all file paths are of the form:
+    # /g/data/if87/datacube/002/S2_MSI_ARD/packaged/YYYY-mm-dd/<granule>
+    # we can simply list all the granules per date and sync them
+    if end_date == 'today':
+        datetime_end = datetime.datetime.today()
+    else:
+        datetime_end = datetime.datetime.strptime(end_date, "%Y-%m-%d")
+
+    # Get list of granules
+    list_of_granules = find_granules(num_days, datetime_end)
+
+    LOG.info("Found {} files to process".format(len(list_of_granules)))
+
+    # For each granule, sync it if it needs syncing
+    if len(list_of_granules) > 0:
+        for granule in list_of_granules:
```

Comment:
We can use a `ThreadPool` here to parallelise, should get close to N speedup for reasonable values of N.